### PR TITLE
Dotnet6 fix

### PR DESCRIPTION
**Pull request by leighmetzroth**

### CODE REVIEW

**Comment by jzabroski** [comment]
Getting closer. There seems to be a problem with test registration. It might only be with Ubuntu (although i dont see why) https://github.com/toddams/RazorLight/runs/5135227279?check_suite_focus=true#step:9:434

What OS do you use?

File: tests/RazorLight.Tests/RazorLight.Tests.csproj [hunk 1]
```diff
@@ -49,18 +45,6 @@
...
-    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.8.3" />
-    <PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="16.8.3" />
-    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="2.1.0" />
-    <PackageReference Include="Microsoft.Extensions.Hosting" Version="2.1.0" />
-  </ItemGroup>
-
   <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.0'">
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.2.0" />
     <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="2.1.0" />
```
[the remark this answers is not in the document]
Comment (jzabroski) [reply tests/RazorLight.Tests/RazorLight.Tests.csproj:50]: @jacodv I guess the question is, which branch should have @leighmetzroth used?  You had quite a few.  Even I am not 100% sure which one was the 'final' a-ha moment.

[the remark this answers is not in the document]
Comment (jacodv) [reply tests/RazorLight.Tests/RazorLight.Tests.csproj:50]: Ok, Let me open the project and do a quick refresh on where I was and what I did.  Then I will revert shortly.

[the remark this answers is not in the document]
Comment (jacodv) [reply tests/RazorLight.Tests/RazorLight.Tests.csproj:50]: @jzabroski  I can confirm that the latest branch, *branch_jdevil_Dotnet6_WebApi6*, is the correct branch.  I use my own solution due to full framework requirements on some projects.  This was the reason that I did not create a pull request.  See detail below...

I have one un-commited change: (Only necessary for a ReSharper Test Runner workaround)
image

I do however use my own solution in the branch - *RazorLight.Jdv.sln*.  This branch does not load the Full framework projects.  I don't have it installed on my PC
image

All tests pass on my branch in VS Tests as well as ReSharper Tests
image

[the remark this answers is not in the document]
Comment (jacodv) [reply tests/RazorLight.Tests/RazorLight.Tests.csproj:50]: Below is the project file for RazorLight.Tests.csproj in the latest source

`<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
	<TargetFrameworks>netcoreapp2.0;netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
    <IsPackable>false</IsPackable>
    <PreserveCompilationContext>true</PreserveCompilationContext>
    <DefineConstants>$(DefineConstants);SOME_TEST_DEFINE</DefineConstants>
    <GenerateRuntimeConfigDevFile>true</GenerateRuntimeConfigDevFile>
  </PropertyGroup>

  <PropertyGroup>
    <MdTargetDir>$(SolutionDir)</MdTargetDir>
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
    <Optimize>false</Optimize>
    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
  </PropertyGroup>

  <ItemGroup>
    <None Remove="Assets\Embedded\*.cshtml" />
    <EmbeddedResource Include="Assets\Embedded\*.cshtml" />
  </ItemGroup>

  <ItemGroup>
    <None Remove="Assets\Files\*.cshtml" />
    <Content Include="Assets\Files\*.cshtml">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="MarkdownSnippets.MsBuild" Version="22.0.3" />
    <PackageReference Include="Moq" Version="4.7.99" />
    <PackageReference Include="Pose" Version="1.2.1" />
    <PackageReference Include="Verify.Xunit" Version="10.0.0-beta.3" />
    <PackageReference Include="xunit" Version="2.4.1" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.0'">
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.2.0" />
    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="2.1.0" />
    <PackageReference Include="Microsoft.Extensions.Hosting" Version="2.1.0" />
  </ItemGroup>

  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="3.1.5" />
    <PackageReference Include="Microsoft.Extensions.Hosting" Version="3.1.5" />
  </ItemGroup>

  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0'">
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.8.0" />
    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="5.0.0" />
    <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
  </ItemGroup>

  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.0.0" />
    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.Hosting" Version="6.0.0" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\src\RazorLight\RazorLight.csproj" />
  </ItemGroup>

</Project>
`

[the remark this answers is not in the document]
Comment (leighmetzroth) [reply tests/RazorLight.Tests/RazorLight.Tests.csproj:50]: Yeah, I think you're right that it was the missing `<ItemGroup>` tag for .NET 6.
This line is where you can see it couldn't find the `testhost.dll` file:
https://github.com/toddams/RazorLight/runs/5135227279?check_suite_focus=true#step:9:468

I have just pushed the change.

**Pull request merged**